lib/fxpay/products.js

```diff
@@ -8,38 +8,66 @@
   var utils = fxpay.getattr('utils');
   var API = fxpay.getattr('api').API;
 
-  exports.all = function products_all(callback) {
-    var promise = new Promise(function(resolve, reject) {
-      var allProducts = [];
 
-      var api = new API(settings.apiUrlBase);
-      var origin = utils.getSelfOrigin();
-      if (!origin) {
-        return reject(errors.InvalidApp('an origin is needed to get products'));
-      }
-      origin = encodeURIComponent(origin);
-      var url;
-
-      if (settings.fakeProducts) {
-        settings.log.warn('about to fetch fake products');
-        url = '/payments/stub-in-app-products/';
-      } else {
-        settings.log.info('about to fetch real products for app',
-                          origin);
-        url = '/payments/' + origin + '/in-app/?active=1';
-      }
+  exports.get = function products_get(productId, opt) {
+    return buildProductReceiptMap().then(function() {
```

Comment:
I was thinking it's quite nice to have all thens on the following line. Which makes them more obvious.